src/controls/MyTeams/components/Team/Team.tsx

```diff
@@ -0,0 +1,283 @@
+  props: React.PropsWithChildren<ITeamProps>
+) => {
+  const {
+    team,
+    serviceScope,
+    onSelectedChannel,
+    themeVariant,
+    enablePersonCardInteraction,
+  } = props;
+  const { styleClasses } = getMyTeamsStyles(themeVariant);
+
+  const { getTeamMembers, getTeamChannels, getTeamOwners } = useTeams(
+    serviceScope
+  );
+
+  const [state, dispatch] = React.useReducer(teamsReducer, initialState);
+
+  const onClickChannel = React.useCallback(
+    (
+      ev: React.MouseEvent<HTMLElement, MouseEvent>,
+      item: IContextualMenuItem
+    ) => {
+      onSelectedChannel
+        ? onSelectedChannel(item.data.teamId, item.key)
+        : window.open(item.data.webUrl);
+    },
+    [onSelectedChannel]
+  );
+
+  const _renderItem = React.useCallback(
+    (itemProps: IContextualMenuItemProps) => {
+      return (
+        <Stack
+          horizontalAlign="start"
+          horizontal
+          verticalAlign="center"
+          tokens={{ childrenGap: 10 }}
+          styles={{ root: { paddingLeft: 10, paddingRight: 10 } }}
+        >
+          <FontIcon iconName={itemProps.item.iconProps.iconName} />
+          <Text variant="small" block>
+            {itemProps.item.text}{" "}
+          </Text>
+          {itemProps.item.secondaryText === "isFavourite" ? (
+            <Stack horizontal horizontalAlign="end" verticalAlign="center">
+              <FontIcon iconName="FavoriteStarFill" />
+            </Stack>
+          ) : null}
+        </Stack>
+      );
+    },
+    []
+  );
+
+  React.useEffect(() => {
+    (async () => {
+      try {
+        dispatch({
+          type: ETeamTypes.SET_HAS_ERROR,
+          payload: false,
+        });
+        dispatch({
+          type: ETeamTypes.SET_IS_LOADING,
+          payload: true,
+        });
+        dispatch({
+          type: ETeamTypes.SET_MESSAGE,
+          payload: undefined,
+        });
+
+        const _members: ITeamMenber[] = await getTeamMembers(team.id);
+        const teamOwners: ITeamMenber[] = await getTeamOwners(team.id);
+        let _renderOwners: string[] = [];
+        for (const teamOwner of teamOwners) {
+          _renderOwners.push(teamOwner.displayName);
+        }
+        dispatch({
+          type: ETeamTypes.SET_TEAM_OWNERS,
+          payload: _renderOwners.join(","),
+        });
+
+        let publicChannels: IContextualMenuItem[] = [];
+        let privateChannels: IContextualMenuItem[] = [];
+        let _renderMembers: string[] = [];
+        let allChannels: IContextualMenuItem[];
+
+        for (const teamMember of _members) {
+          _renderMembers.push(teamMember.userId);
+        }
+        dispatch({
+          type: ETeamTypes.SET_TEAM_MEMBERS,
+          payload: _renderMembers,
+        });
+
+        // get Channels
+        const channels = await getTeamChannels(team.id);
+
+        publicChannels.push({
+          itemType: ContextualMenuItemType.Header,
+          key: "public",
+          text: "Public Channels",
```

Comment:
@joaojmendes can you please update to use a loc string? It's fine if you can't do it and I will do it after the merge

Reply:
done